### PR TITLE
Random Access Test Pattern Generators

### PR DESCRIPTION
**Pull request by jamesba**
Job: https://www.pivotaltracker.com/story/show/172645870

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -8,7 +8,7 @@ envlist = py36
 
 [testenv]
 commands =
-    python -m unittest discover -s tests -p test_*.py
+    python -m {posargs:unittest discover -s tests -p test_*.py}
```
Comment (philipnbbc) [inline tox.ini:11]: :+1: Easier than  having to modify the tox file to try things, run a specific test only.

File: mediagrains/patterngenerators/audio/tone.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+            channels=self.channels,
+            rate=self.rate,
+            sample_rate=self.sample_rate)
+
+        ag.data = bytearray(self._get_samples(offs)[:ag.expected_length])
+
+        return ag
+
+
+def Tone1K(
```
Comment (philipnbbc) [inline mediagrains/patterngenerators/audio/tone.py:184]: Should this be dropped for simplicity? I think it was created as a stop gap before the more general class.

File: tox.ini (hunk 1 above)
Comment (jamesba) [reply tox.ini:11]: Perhaps this should go into commontooling?

File: mediagrains/patterngenerators/audio/tone.py (hunk 2 above)
Comment (jamesba) [reply mediagrains/patterngenerators/audio/tone.py:184]: I think it hurts nothing to keep it.

File: tox.ini (hunk 1 above)
Comment (philipnbbc) [reply tox.ini:11]: I think that would be useful, yes.

**Pull request merged**